brave/brave-core
# feat(wallet): Portfolio Filters and Display Settings

## Description 
Introduces a `Portfolio Filters and Display Settings` modal.

Settings include:
- `Sort assets` dropdown (High to Low, Low to High, A to Z, Z to A)
- `Hide small balances` toggle (will hide balance with a value less than $1)
- `Select networks` filter (Assets of checked networks will be displayed in portfolio, note: by default all `test networks` will be unchecked, if a user checks a `test network` the value will be included in the `portfolio`.)
- `Select accounts` filter (Balances will be reflected by what `accounts` are checked.)

The `Edit Visible Assets` button is no longer at the bottom of the page but is now a new button to the left of the `Portfolio Filters and Display Settings` button at the top right of the portfolio assets list.


Resolves <https://github.com/brave/brave-browser/issues/30496>



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

### Testing `Sort assets`:
1. Go to the `Portfolio` page and open the `Filters and display settings`
2. Change the `Sort assets` option and click `Save changes`
3. The portfolio should reflect your selected option
4. Test this for all options.

https://github.com/brave/brave-core/assets/40611140/b470e8a3-ee75-44ec-9dc2-562dd368956e

### Testing `Hide small balances`:
1. Go to the `Portfolio` page and open the `Filters and display settings`
2. Toggle on the `Hide small balances` toggle and click `Save changes`
3. The portfolio should hide assets with a balance less than $1

https://github.com/brave/brave-core/assets/40611140/f2ac399d-f3e5-4848-8279-36e4b9bbf2e5

### Testing `Select networks`:
1. Go to the `Portfolio` page and open the `Filters and display settings`
2. Check or Uncheck `networks` and click `Save changes`
3. The portfolio should reflect your `Selected networks` (note: by default all `test networks` will be unchecked, if a user checks a `test network` the value will be included in the `portfolio`).
4. Test the `Select all` and `Deselect all` networks button.

https://github.com/brave/brave-core/assets/40611140/dc9947f3-0269-4dd6-89af-02941a498041

### Testing `Select accounts`:
1. Go to the `Portfolio` page and open the `Filters and display settings`
2. Check or Uncheck `accounts` and click `Save changes`
3. The portfolio should reflect your `Selected accounts` (note: `Selected Networks` takes a higher priority and assets will remain visible but will show a `0` balance if associated accounts are hidden.)
4. Test the `Select all` and `Deselect all` accounts button.

https://github.com/brave/brave-core/assets/40611140/f4a05c2e-c201-45c1-b5ed-340b5cea2055
